elasticsearch/src/test/java/org/hibernate/search/elasticsearch/test/bridge/MapAsInnerObjectFieldBridge.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.elasticsearch.test.bridge;
+
+import java.util.Map;
+
+import org.apache.lucene.document.Document;
+import org.hibernate.search.bridge.LuceneOptions;
+import org.hibernate.search.bridge.MetadataProvidingFieldBridge;
+import org.hibernate.search.bridge.spi.FieldMetadataBuilder;
+import org.hibernate.search.bridge.spi.FieldType;
+import org.hibernate.search.elasticsearch.bridge.builtin.impl.Elasticsearch;
+import org.hibernate.search.elasticsearch.schema.impl.model.DynamicType;
+
+/**
+ * @author Davide D'Alto
+ */
+public class MapAsInnerObjectFieldBridge implements MetadataProvidingFieldBridge {
+
+	@Override
+	public void set(String name, Object value, Document document, LuceneOptions luceneOptions) {
+		if ( !( value instanceof Map ) ) {
+			throw new IllegalArgumentException( "This field can only be applied on a Map type field" );
+		}
+		else {
+			Map<Object, Object> userValue = (Map) value;
+			for ( Map.Entry<Object, Object> e : userValue.entrySet() ) {
+				setField( name, String.valueOf( e.getKey() ), String.valueOf( e.getValue() ), document, luceneOptions );
+			}
+		}
+	}
+
+	private void setField(String fieldPrefix, String key, String value, Document document, LuceneOptions luceneOptions) {
+		luceneOptions.addFieldToDocument( fieldPrefix + "." + key, value, document );
+	}
+
+	@Override
+	public void configureFieldMetadata(String name, FieldMetadataBuilder builder) {
+		builder.field( "dynamicField", FieldType.OBJECT )
+				.mappedOn( Elasticsearch.class )
+				.dynamic( DynamicType.TRUE );
+
+		builder.field( "dynamicStrict", FieldType.OBJECT );
```

Comment:
I'm a bit perplexed: "dynamicStrict" does not match any field name... I wonder why this works.

Anyway... Maybe there should be two different bridges here? It feels a bit strange that the `name` parameter in `configureFieldMetadata` is ignored.

Same as before: I agree it can work as is, but it's more in case someone copy/pastes this code.